atlasaction/action.go

```diff
@@ -40,3 +54,63 @@ func MigrateApply(ctx context.Context, client *atlasexec.Client, act *githubacti
 	act.Infof("Run complete: +%v", run)
 	return nil
 }
+
+// MigratePush runs the Github Action for "ariga/atlas-action/migrate/push"
+func MigratePush(ctx context.Context, client *atlasexec.Client, act *githubactions.Action) error {
+	ghContext, err := createContext(act)
+	if err != nil {
+		return fmt.Errorf("failed to read github metadata: %w", err)
+	}
+	buf, err := json.Marshal(ghContext)
+	if err != nil {
+		return fmt.Errorf("failed to create MigratePushParams: %w", err)
+	}
+	params := &atlasexec.MigratePushParams{
+		Name:      act.GetInput("dir-name"),
+		DirURL:    act.GetInput("dir"),
+		DevURL:    act.GetInput("dev-url"),
+		Context:   string(buf),
+		ConfigURL: act.GetInput("config"),
+		Env:       act.GetInput("env"),
+	}
+	_, err = client.MigratePush(context.Background(), params)
+	if err != nil {
+		return fmt.Errorf("failed to push directory: %v", err)
+	}
+	tag := act.GetInput("tag")
+	if tag != "" {
+		params.Tag = tag
+	} else {
+		params.Tag = ghContext.Commit
+	}
+	resp, err := client.MigratePush(context.Background(), params)
+	if err != nil {
+		return fmt.Errorf("failed to push dir tag: %w", err)
+	}
+	act.SetOutput("url", resp)
+	act.Infof("Uploaded migration dir %q to Atlas Cloud\n", params.Name)
+	return nil
+}
+
+func createContext(github *githubactions.Action) (*ContextInput, error) {
+	ghContext, err := github.Context()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load action context: %w", err)
+	}
+	var ev struct {
+		HeadCommit struct {
+			URL string `mapstructure:"url"`
+		} `mapstructure:"head_commit"`
+		Ref string `mapstructure:"ref"`
+	}
+	if err := mapstructure.Decode(ghContext.Event, &ev); err != nil {
+		return nil, fmt.Errorf("failed to parse push event: %v", err)
+	}
+	return &ContextInput{
+		Repo:   ghContext.Repository,
+		Branch: ghContext.RefName,
+		Commit: github.Getenv("GITHUB_SHA"),
+		Path:   github.GetInput("dir"),
```

Comment:
Not sure about using `dir` as `Path` here:
1. We expect a URL format
2. Are we guaranteed that it is relative to the repo root?

Reply:
1. True. Should I invest time fixing it?
2. I'm not sure, but I think that yes since the working directory input is only used by the SDK. In a short discussion we had about this at the office @a8m said not to invest time in it, so it's not very well tested